x/ccv/consumer/keeper/keeper.go

```diff
@@ -206,6 +207,41 @@ func (k Keeper) DeletePendingChanges(ctx sdk.Context) {
 	store.Delete(types.PendingChangesKey())
 }
 
+func (k Keeper) IsPreCCV(ctx sdk.Context) bool {
+	store := ctx.KVStore(k.storeKey)
+	bz := store.Get(types.PreCCVKey())
+	if bz != nil {
+		return true
+	}
+	return false
+}
+
+func (k Keeper) SetPreCCV(ctx sdk.Context, state *types.GenesisState) {
+	initialValSet := types.GenesisState{
```

Comment:
PreCCV is just a flag correct? Seems like it should be set from `GenesisState.preCCV`. Not sure why we're storing a val set here

Reply:
PreCCV is the flag, but when preCCV state ends, it should use genesis validator set, and therefore, it's putting initial validator set there. I think it would be better to use this rather than adding one more key for initial validator set. Since it's pretty much connected with each other. This could be separated though.